jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ConditionalHandler.java

```diff
@@ -0,0 +1,811 @@
+    private final IncludeExcludeSet<Predicate<Request>, Request> _predicates = new IncludeExcludeSet<>(PredicateSet.class);
+    private Predicate<Request> _handlePredicate;
+
+    private ConditionalHandler()
+    {
+        this(false, null);
+    }
+
+    private ConditionalHandler(Handler nextHandler)
+    {
+        this(false, nextHandler);
+    }
+
+    private ConditionalHandler(boolean dynamic, Handler nextHandler)
+    {
+        super(dynamic, nextHandler);
+    }
+
+    /**
+     * Clear all inclusions and exclusions.
+     */
+    public void clear()
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+        _methods.clear();
+        _pathSpecs.clear();
+        _predicates.clear();
+    }
+
+    IncludeExclude<String> getMethods()
+    {
+        // Used only for testing
+        return _methods;
+    }
+
+    IncludeExclude<String> getPathSpecs()
+    {
+        // Used only for testing
+        return _pathSpecs;
+    }
+
+    IncludeExcludeSet<Predicate<Request>, Request> getPredicates()
+    {
+        // Used only for testing
+        return _predicates;
+    }
+
+    /**
+     * Include {@link Request#getMethod() method}s in the conditions to be met
+     * @param methods The exact case-sensitive method name
+     */
+    public void includeMethod(String... methods)
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+        _methods.include(methods);
+    }
+
+    /**
+     * Exclude {@link Request#getMethod() method}s in the conditions to be met
+     * @param methods The exact case-sensitive method name
+     */
+    public void excludeMethod(String... methods)
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+        _methods.exclude(methods);
+    }
+
+    /**
+     * Include {@link PathSpec}s in the conditions to be met
+     * @param paths The {@link PathSpec}s that are tested against the {@link Request#getPathInContext(Request) pathInContext}.
+     */
+    public void include(PathSpec... paths)
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+
+        for (PathSpec p : paths)
+            ((PathSpecSet)_pathSpecs.getIncluded()).add(p);
+    }
+
+    /**
+     * Exclude {@link PathSpec}s in the conditions to be met
+     * @param paths The {@link PathSpec}s that are tested against the {@link Request#getPathInContext(Request) pathInContext}.
+     */
+    public void exclude(PathSpec... paths)
+    {
+        if (isStarted())
+            throw new IllegalStateException(getState());
+        for (PathSpec p : paths)
+            ((PathSpecSet)_pathSpecs.getExcluded()).add(p);
+    }
+
+    /**
+     * Include {@link PathSpec}s in the conditions to be met
+     * @param paths String representations of {@link PathSpec}s that are
+     *              tested against the {@link Request#getPathInContext(Request) pathInContext}.
```

Comment:
This javadoc contains extra spaces that can be cleaned up. Ditto for the following methods.

Reply:
They are not extra spaces, they are indents.   It is a common style.
whatever. will remove.